r/tests/testthat/test-csv.R

```diff
@@ -317,7 +317,7 @@ test_that("Write a CSV file with different batch sizes", {
 test_that("Write a CSV file with invalid input type", {
   expect_error(
     write_csv_arrow(Array$create(1:5), csv_file),
-    regexp = 'x must be a "ArrowTabular"'
+    regexp = "Cannot write to CSV. 'x' must be an object of class 'data.frame', 'RecordBatch', or 'Table', not 'Array'"
```

Comment:
This is also not strictly about this pull request, but I believe the 📦 would benefit from switching to testthat 3rd edition `expect_snaphot()` ... https://testthat.r-lib.org/articles/third-edition.html

Reply:
Sounds good - Jon opened a ticket to do that